python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -1121,10 +1094,15 @@ def getJobObject(self):
         Unknown: Why is this a GangaObject method and not a Node method?
         """
         from Ganga.GPIDev.Lib.Job import Job
-        r = self._getRoot(cond=lambda o: isinstance(o, Job))
-        if not isinstance(r, Job):
-            raise AssertionError('no job associated with object ' + repr(self))
-        return r
+        if self._getParent():
+            r = self._getRoot(cond=lambda o: isinstance(o, Job))
+            if not isinstance(r, Job):
+                raise AssertionError('No Job associated with object instead root=\'%s\' for \'%s\'' % (repr(self), type(r)))
+            return r
+        else:
+            if isinstance(self, Job):
+                return self
+        raise AssertionError('No Parent associated with object instead root=\'%s\' for \'%s\'' % (repr(self), str(None)))
```

Comment:
The fact that it's not clear whether you don't have a parent associated with an object or whether that parent is a Job. (I was playing with objects in a Box repo and this confused/bugged me as to why the error made no sense when I was testing something). Having no parent and having no Job object should look different to each other imo. I'll move this into another PR in this case.